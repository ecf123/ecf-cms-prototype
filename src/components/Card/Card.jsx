import "./Card.scss";
import iconArrow from "../../assets/arrow-upper-right-black.svg";
const Card = ({ img, dateOrTime, title, cardInfo, links }) => {
  typeof links === "string" ? (links = [links]) : links;
  const linksArr = links?.map((link, index) => {
    if (index % 2 == 0) {
      return (
        <li key={index} className="card__link card__link--purple">
          {link}
        </li>
      );
    } else {
      return (
        <li key={index} className="card__link card__link--darkpurple">
          {link}
        </li>
      );
    }
  });
  return (
    <div className="card">
      <img className="card__image" src={img} alt="Card Image" />
      <p className="card__date">{dateOrTime}</p>
      <div className="card__header">
        <h2 className="card__title">{title}</h2>
        <img className="card__icon" src={iconArrow} alt="Arrow Icon" />
      </div>
      <p className="card__info">{cardInfo}</p>
      <ul className="card__links">{linksArr}</ul>
    </div>
  );
};
export default Card;