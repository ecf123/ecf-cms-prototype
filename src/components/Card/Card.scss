@use "../../styles/variables" as *;

.card {
  font-family: $font-secondary;
  max-width: 26.7rem;
  height: 33rem;
  font-style: normal;

  &__image {
    width: 100%;
    height: auto;
  }

  &__date {
    font-weight: $font-weight-regular;
    font-size: $font-size-extra-small;
    line-height: 1.2rem;
    color: $color-teal;
  }

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    text-decoration: none;

    .card__icon {
      margin-top: 0.6rem;
    }
    .card__title {
      font-weight: $font-weight-regular;
      font-size: $font-size-medium;
      line-height: 2rem;
      color: $color-black;
    }
  }

  &__info {
    font-weight: $font-weight-regular;
    font-size: $font-size-small;
    line-height: 1.5rem;
    margin-bottom: 1.6rem;
    color: $color-teal;
  }

  &__links {
    font-weight: $font-weight-regular;
    font-size: $font-size-extra-small;
    line-height: 1.2rem;
    list-style: none;
    display: flex;
    flex-wrap: wrap;
  }

  &__link {
    margin: 0 0.6rem;
    border-radius: 0.6rem;
    padding: 0 0.6rem;

    &--purple {
      color: $color-purple;
      background-color: #f9f5ff;
    }

    &--darkpurple {
      color: $color-purple-dark;
      background-color: #f8f9fc;
    }
  }
}
