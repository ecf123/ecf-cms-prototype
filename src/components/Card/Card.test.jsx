import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";

import Card from "./Card";

it("should render the card image", () => {
  //Arrange
  render(<Card imgDescription={"Card Image"} />);
  //Act
  const image = screen.getByAltText("Card Image");
  //Assert
  expect(image).toBeInTheDocument();
});

it("should display the date or time on the screen", () => {
  //Arrange
  render(<Card dateOrTime="22/01/2022" />);
  //Act
  const date = screen.queryByText("22/01/2022");
  //Assert
  expect(date).toBeInTheDocument();
});

it("should display the header of the card on the screen", () => {
  //Arrange
  render(<Card title="Test Title" />);
  //Act
  const header = screen.queryByText("Test Title");
  //Assert
  expect(header).toBeInTheDocument();
});

it("should display the links on the screen", () => {
  //Arrange
  render(<Card links={["linkOne"]} />);
  //Act
  const link = screen.getByText("linkOne");
  //Assert
  expect(link).toBeInTheDocument();
});

it("should display the article paragraph on the screen", () => {
  //Arrange
  render(<Card cardInfo="Test card" />);
  //Act
  const paragraph = screen.getByText("Test card");
  //Assert
  expect(paragraph).toBeInTheDocument();
});

it("should render the arrow icon", () => {
  //Arrange
  render(<Card />);
  //Act
  const image = screen.getByAltText("Arrow Icon");
  //Assert
  expect(image).toBeInTheDocument();
});

it("Should render the card with no props", () => {
  const { container } = customRender(<Card />);
  expect(container).toMatchSnapshot();
});

it("Should render the card with props", () => {
  const { container } = customRender(
    <Card
      img="../../assets/for-hire-image.svg"
      dateOrTime="18:25"
      title="Card"
      cardInfo="lorem ipsum"
      links={["one", "two", "three"]}
    />
  );
  expect(container).toMatchSnapshot();
});
