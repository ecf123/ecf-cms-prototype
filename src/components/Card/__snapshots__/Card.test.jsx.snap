// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the card with no props 1`] = `
<div>
  <div
    class="card"
  >
    <a
      href="/*"
    >
      <img
        class="card__image"
      />
    </a>
    <p
      class="card__date"
    />
    <a
      class="card__header"
      href="/*"
    >
      <h2
        class="card__title"
      />
      <img
        alt="Arrow Icon"
        class="card__icon"
        src="arrow-upper-right-black.svg"
      />
    </a>
    <p
      class="card__info"
    />
    <ul
      class="categories-menu"
    />
  </div>
</div>
`;

exports[`Should render the card with props 1`] = `
<div>
  <div
    class="card"
  >
    <a
      href="/Card"
    >
      <img
        class="card__image"
        src="../../assets/for-hire-image.svg"
      />
    </a>
    <p
      class="card__date"
    >
      18:25
    </p>
    <a
      class="card__header"
      href="/Card"
    >
      <h2
        class="card__title"
      >
        Card
      </h2>
      <img
        alt="Arrow Icon"
        class="card__icon"
        src="arrow-upper-right-black.svg"
      />
    </a>
    <p
      class="card__info"
    >
      lorem ipsum
    </p>
    <ul
      class="categories-menu"
    >
      <li
        class="categories-menu__item categories-menu__item--purple"
      >
        one
      </li>
      <li
        class="categories-menu__item categories-menu__item--darkpurple"
      >
        two
      </li>
      <li
        class="categories-menu__item categories-menu__item--purple"
      >
        three
      </li>
    </ul>
  </div>
</div>
`;
